quick_select/quick_select.py

```diff
@@ -0,0 +1,60 @@
+def partition(array, start, end):
+    """
+    Perform Partition Operation on array a.
+    Time Complexity: O(nLogn)
+    Auxiliary Space: O(n)
+    :param a: Iterable of elements
+    :param start: pivot value for array
+    :param end: right limit of array
+    :return: return i value for function, used in partitioning of array.
+    """
+    i = start - 1
+    pivot = array[end]
+    for j in range(start, end):
+        if array[j] <= pivot:
+            i += 1
+            array[i], array[j] = array[j], array[i]
+    i += 1
+    array[i], array[end] = array[end], array[i]
+    return i
+
+
+def quick_select(array, k):
```

Comment:
The cyclomatic complexity can be decreased, take a look down below the comment on line 56.